packet/adaptationfield.go

```diff
@@ -0,0 +1,578 @@
+/*
+MIT License
+
+Copyright 2016 Comcast Cable Communications Management, LLC
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package packet
+
+import (
+	"github.com/Comcast/gots"
+)
+
+// NewPacket creates a new packet with a Null ID, sync byte, and with the adaptation field control set to payload only.
+// This function is error free.
+func NewAdaptationField() AdaptationField {
+	p := NewPacket()
+	p.SetAdaptationFieldControl(AdaptationFieldFlag)
+	af, _ := p.AdaptationField()
+	return af
+}
+
+// setBit sets a bit in the adaptation field
+// index is the byte in the packet where the bit is located
+// mask is the bitmask that has that bit set to one and the rest of the bits set to zero
+// value is the value that the bit will be set to
+func (af AdaptationField) setBit(index int, mask byte, value bool) {
```

Comment:
This also feels a bit redundant, but I suppose this makes sense since packet is `type [188]byte`